services/swagger.js

```diff
@@ -131,6 +134,34 @@ exports.addFormat = function (format, validationFunction) {
     tv4.addFormat(format, validationFunction);
 };
 
+function overrideInheritedProperties(definitions) {
+    Object.keys(definitions).forEach(function (defn) {
+        var props = {};
+        if (definitions[defn].hasOwnProperty('allOf')) {
+            for (var i = 0; i < definitions[defn].allOf.length; i++) {
+                //have to clone so that inherited definitions are not affected
+                definitions[defn].allOf[i] = swaggerUtil.deepClone(definitions[defn].allOf[i]);
+                overrideProps(definitions[defn].allOf[i], props);
+            }
+            definitions[defn].properties = props;
+        }
+    });
+}
+
+function overrideProps(item, props) {
+    if (item.hasOwnProperty('allOf')) {
+        item.allOf.forEach(function (subItem) {
+            overrideProps(subItem, props);
+        });
+    }
+    if (item.properties) {
+        Object.keys(item.properties).forEach(function (key) {
+            props[key] = item.properties[key];
+        });
+        delete item.properties;
```

Comment:
instead of deleting properties and so that we can keep the 'allOf' structure, what if we simply generate the resulting object (handling nesting appropriately), the simple algorithm would be:

1. `_.merge({}, theAllOffArray)`